src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -37,16 +38,27 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
             Document? document = null;
 
             // If we were given a document, find it in whichever workspace it exists in
-            if (textDocument is not null)
+            if (textDocument is null)
+            {
+                _logger.TraceInformation("Request contained no document id");
+            }
+            else
             {
+                _logger.TraceInformation($"Finding document corresponding to: {textDocument.Uri}");
+
                 // There are multiple possible solutions that we could be interested in, so we need to find the document
                 // first and then get the solution from there. If we're not given a document, this will return the default
                 // solution
                 document = FindDocument(lspWorkspaceRegistrationService, textDocument, clientName);
 
-                if (document is not null)
+                if (document is null)
+                {
+                    _logger.TraceWarning($"Found no corresponding document");
+                }
+                else
                 {
                     // Where ever the document came from, thats the "main" solution for this request
+                    _logger.TraceInformation($"Found document: {document.FilePath}");
```

Comment:
Consider adding workspace kind to this, in case we found the document in misc files but expected it in regular?

Reply:
done.